internal/xds/translator/translator.go

```diff
@@ -896,7 +896,7 @@ func buildXdsUpstreamTLSSocketWthCert(tlsConfig *ir.TLSUpstreamConfig) (*corev3.
 		tlsCtx.CommonTlsContext.TlsParams = tlsParams
 	}
 
-	if len(tlsConfig.ALPNProtocols) > 0 {
+	if tlsConfig.ALPNProtocols != nil {
```

Comment:
I think it would be clearer to change `buildALPNProtocols` to return an empty `[]string` if `tlsConfig.ALPNProtocols` is nil. There's no need to check the length of the `tlsConfig.ALPNProtocols` array here - it's already checked in the `buildALPNProtocols` function.